tests/utils/dockerized.py

```diff
@@ -0,0 +1,44 @@
+import docker
+import pytest
+from pathlib2 import Path
+
+
+def dockerized_case(builder_cls, file_path):
+    """Decorator that makes decorated test case run inside docker container."""
+    def dockerized_real(f):
+        root = Path(__file__).parent.parent.parent
+        rel_path = Path("agent_source") / Path(file_path).relative_to(root)
+
+        command = "python -u -m pytest {0}::{1} -s --no-dockerize".format(
+            rel_path,
+            f.__name__
+        )
+
+        def wrapper(*args, **kwargs):
+            no_dockerize = pytest.config.getoption("--no-dockerize")
+            if no_dockerize:
+                result = f(*args, **kwargs)
+                return result
+
+            builder = builder_cls()
+            builder.build()
```

Comment:
Some of those commands (Building docker image), etc. take quite a while so in those scenarios we may want to run py.test with ``-s`` flag to output is not captured and printed to stdout - this way we know why tests are still running and not stuck.

Reply:
And ideally we would also periodically print progress inside ``builder.build()`` method, but that's not a blocker for now.

Reply:
Is there any way we could use image cache here?

If we can't, this will be very slow since we will need to re-build the image from scratch on CI during each job run if DLC is disabled...

Reply:
> Some of those commands (Building docker image), etc. take quite a while so in those scenarios we may want to run py.test with -s flag to output is not captured and printed to stdout - this way we know why tests are still running and not stuck.
It prints build output, but it looks like we need to wait until the image is built, and only then we can read the build output. I tried to find the way to read in real-time, but without success :(

> Is there any way we could use image cache here?
Yes, now it can get path to cache from py.test option.